src/hera/task.py

```diff
@@ -720,6 +725,7 @@ def get_container(self) -> Container:
         """
         container_kwargs = {
             "image": self.image,
+            "image_pull_policy": self.image_pull_policy,
```

Comment:
With the proposed change to use `ImagePullPolicy`, this should be:
```suggestion
            "image_pull_policy": self.image_pull_policy.value,
```

Reply:
`get_script_def` above `get_container` also needs the pull policy specified!